src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx

```diff
@@ -46,33 +47,45 @@ export const WalletSelectionScreen = () => {
     onError: async (error) => {
       navigation.setParams({reopen: true})
       if (error instanceof SystemAuthDisabled) {
-        await closeWallet()
-        await showErrorDialog(errorMessages.enableSystemAuthFirst, intl)
-        resetToWalletSelection()
+        systemAuthDisabledError()
       } else if (error instanceof InvalidState) {
-        await closeWallet()
-        await showErrorDialog(errorMessages.walletStateInvalid, intl)
-        resetToWalletSelection()
+        invalidStateError()
       } else if (error instanceof KeysAreInvalid) {
         await showErrorDialog(errorMessages.walletKeysInvalidated, intl)
-        await logout()
+        logout()
       } else {
         throw error
       }
     },
   })
 
+  const {closeWalletWithWalletMeta} = useCloseWalletWithWalletMeta({
+    onError: async () => {
+      await showErrorDialog(errorMessages.walletStateInvalid, intl)
+    },
+    onSuccess: (walletMeta) => {
+      openWallet(walletMeta)
```

Comment:
can you access the latest walletMeta via closure?

Reply:
A refactor has been applied